flag.go

```diff
@@ -610,6 +607,19 @@ func withEnvHint(envVars []string, str string) string {
 	return str + envText
 }
 
+func flagNames(f Flag) []string {
+	ret := []string{}
+
+	name := flagStringField(f, "Name")
+	aliases := flagStringSliceField(f, "Aliases")
+
+	for _, part := range append([]string{name}, aliases...) {
```

Comment:
What do you think about including a comment here so future spelunkers aren't confused as to what is happening here?